### ISSUE TITLE
Support SSL and TLS

### ISSUE THREAD

**Issue by ntt1994** [criteo/kafka-sharp#51]
Hi Criteo, can kafka sharp support ssl and tls? 
Thanks.

**Comment by ychuzevi** [comment]
Hello,
There is no encryption support for now. But the next release should bring authentication (sasl oauth) which is currently under development.

**Comment by ntt1994** [comment]
Thank for reply. I hope the next version is published soon.

**Comment by mrpmorris** [comment]
Has this project been abandoned?

**Comment by ychuzevi** [comment]
Yes. Criteo switched to the official C# driver which is now providing proper performances (and support of all the latest features)

**Comment by mrpmorris** [comment]
Thank you!

May I recommend putting a large notice at the top of readme.md, and marking the repo as frozen?

I see the official library uses async for produce but thread blocking for consume. What are your thoughts/experiences on that?

Thank you for making your source code publicly available btw!

**Comment by julesbovet** [comment]
Hello!
It is indeed blocking for consume, same as the Criteo kafka client (APIs are a bit different because it is event driven, but in the Criteo kafka client, if you were to block in the event, you were actually blocking the consume thread).

From our experience:
- If what you want to do is very lightweight, like incrementing a metric, this can be done on the thread consuming messages
- If you want to do more heavy processing, you should enqueue the messages somewhere (BlockingCollection is a good candidate) to free the consume thread and have other threads do the heavy lifting. One thing to keep it mind though is that you also want to make sure you're not consuming more messages than what your application is actually able to process, hence why BlockingCollection is often used for this scenario.

**Comment by verdie-g** [comment]
> May I recommend putting a large notice at the top of readme.md, and marking the repo as frozen?

Done in https://github.com/criteo/kafka-sharp/pull/57.

**Comment by mrpmorris** [comment]
Thank you for your time!
